E-Hospital/E-Hospital.BusinessLogic/Services/Implementation/RegistrationService.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using AutoMapper;
+using E_Hospital.BLL.Configuration.MappingProfiles;
+using E_Hospital.BLL.Data;
+using E_Hospital.BLL.Utils;
+using E_Hospital.DAL;
+using E_Hospital.DAL.Entities;
+using E_Hospital.DAL.Repositories.Abstraction;
+
+namespace E_Hospital.BLL.Services.Implementation
+{
+    class RegistrationService : IRegistrationService
+    {
+        public RegistrationService()
+        {
+            var unitOfWork = new UnitOfWork();
+            _userRepository = unitOfWork.GetRepository<User>();
+            _roleRepository = unitOfWork.GetRepository<Role>();
+            _doctorRepository = unitOfWork.GetRepository<Doctor>();
+            _patientRepository = unitOfWork.GetRepository<Patient>();
+            _specializationRepository = unitOfWork.GetRepository<Specialization>();
+            _mapper = new Mapper(new MapperConfiguration(cfg => cfg.AddProfile<MappingProfile>()));
+        }            
+
+        public bool RegisterDoctor(DoctorDto doctor)
+        {            
+            if (_userRepository.Single(x => x.Login == doctor.Login) != null)
+                return false;
+
+            string salt = EncryptionUtil.GenerateSalt();
+            string passwd = EncryptionUtil.HashPassword(doctor.Password, salt);
+            var user = new User()
+            {
+                LastName = doctor.LastName,
+                FirstName = doctor.FirstName,
+                Login = doctor.Login,
+                Password = passwd,
+                Salt = salt,
+                PhoneNumber = doctor.PhoneNumber,
+                Role = _roleRepository.Single(x => x.Name == doctor.Role.ToString()),
+            };
+            var _doctor = new Doctor()
```

Comment:
Локальні змінні не починаються із "_"
```suggestion
            var doctor = new Doctor()
```